genienlp/arguments.py

```diff
@@ -186,6 +186,13 @@ def parse_argv(parser):
     parser.add_argument('--eval_set_name', type=str, help='Evaluation dataset name to use during training')
 
     parser.add_argument('--max_output_length', default=150, type=int, help='maximum output length for generation')
+    parser.add_argument(
+        '--min_output_length',
+        default=3,
+        type=int,
+        help='maximum output length for generation; '
+        'default is 3 for most multilingual models: BOS, language code, and one token. otherwise it is 2',
```

Comment:
I think it could be 3 for everything. Originally, I set it to three for `BOS, one token, EOS` (no language code). Would the classification task in your experiments be fixed if you added a [EOS] token to the end of your training example like `[BOS] false [EOS]`?

Reply:
Hold on, why is BOS counted in the generation length? We don't generate BOS, it goes into the input.

Reply:
@gcampax  Have you forgotten what you wrote [here](https://github.com/stanford-oval/genienlp/blob/e4f9c80fc21030dad0137b05fd8f40d6d3830456/genienlp/models/transformer_seq2seq.py#L128) ;p 
All pretrained BART checkpoints generate `[BOS] X [EOS]`. For bart-large I think you did some experiments and decided it works better if we finetune without [BOS] in the output. It's generally fine since we finetune and model learns not to generate [BOS].
MBART generates `X [EOS] [lang_code]`. MT5/ T5 generates `X [EOS]`. 
Before introducing multilingual models, min_output_length was hard-coded to 2. After becoming multilingual I made it 3. 
Now trying T5 realized we should do 2 but for backward compatibility with previously trained models, I think we should keep the default 3 and pass what we want for other models.
If we pass the wrong one, it will be caught later in the length exceptions I added in [this commit ](https://github.com/stanford-oval/genienlp/pull/234/commits/c709f72120ed7a890a379645382942de0f07d020)